sw/device/lib/base/mmio.h

```diff
@@ -139,10 +139,14 @@ inline void mmio_region_write32(mmio_region_t base, ptrdiff_t offset,
  *
  * Instead of containing a volatile pointer, mmio_region_t becomes a `void *`
  * when `-DMOCK_MMIO` is enabled. This makes it incompatible with the non-mock
- * version of `mmio_region_t`, which prevents users from being able to access
- * the pointer inside.
+ * version of `mmio_region_t`, the users can use `base` to fake up the base
+ * address in the unittests (useful for the cases when `mmio_region_t` is used
+ * for address calculation or input to `memcpy`).
  */
-typedef struct mmio_region { void *mock; } mmio_region_t;
+typedef struct mmio_region {
+  void *mock;
+  void *base;
```

Comment:
We can roll back the mmio change, as we will move to using the mmio version of memcpy, meaning we don't need `base` any more.

Reply:
Done.